manager/manager.go

```diff
@@ -237,6 +237,12 @@ type manager struct {
 
 // Start the container manager.
 func (self *manager) Start() error {
+	//add-by fzu-huang clean the factories which is a external object.
+	//Usually we just run only one manager.So every time we start a manager
+	//we have to clear old handlerfactories, otherwise, new handlerfactories
+	//will be in behand of old factories. and won't be called.
+	container.ClearContainerHandlerFactories()
```

Comment:
If there is another manager running, this will break it. I think it's better to just return an error in this case, something like `"cannot start manager: container factories already registered"`, and leave it up to the caller to call `container.ClearContainerHandlerFactories()` if this was really intentional.

Ideally we would just move away from the global registry.